src/ngAnimate/animateQueue.js

```diff
@@ -579,6 +586,11 @@ var $$AnimateQueueProvider = ['$animateProvider', function($animateProvider) {
       var animateChildren;
       var elementDisabled = disabledElementsLookup.get(getDomNode(element));
 
+      if (elementDisabled === true) {
+        // If animations on the element have been disabled explicitly, no need to traverse the parents
+        return false;
+      }
```

Comment:
AFAICT, `elementDisabled` could be still set to `false` on [line 626][1], so this change is either invalid or also a fix (in addition to perf improvement). I believe it is the latter (in which case a test should be added).

[1]: https://github.com/angular/angular.js/pull/13784/files#diff-c15d87c15661cba4a721501ba4fda5b2R626